pip/vcs/git.py

```diff
@@ -134,7 +134,7 @@ def obtain(self, dest):
             rev_options = [rev]
             rev_display = ' (to %s)' % rev
         else:
-            rev_options = ['origin/master']
+            rev_options = ['refs/remotes/origin/HEAD']
```

Comment:
One more thing: my impression from digging into pip's code base and following some of the issues and conversation is that while most of the testing in the past has been focused on the end-to-end tests, there is some movement and desire to move towards finer-grained tests (one of the reasons being the slowness you mentioned).

So I think there's value in thinking about how we can better leverage finer-grained tests, even if the patterns aren't yet widely established.